engine/src/main/java/io/camunda/zeebe/engine/api/ProcessingResultBuilder.java

```diff
@@ -7,13 +7,48 @@
  */
 package io.camunda.zeebe.engine.api;
 
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+
+/**
+ * Low level interface to capture the processing result. On engine side, this will probably be
+ * decorated with Writers or something like that
+ */
 public interface ProcessingResultBuilder {
 
+  /** Appends a record to the result */
+  ProcessingResultBuilder appendRecord(
+      final long key,
+      final RecordType type,
+      final Intent intent,
+      final RejectionType rejectionType,
+      final String rejectionReason,
+      final RecordValue value);
+
+  /** Sets the response for the result; will be overwritten if called more than once */
+  ProcessingResultBuilder withResponse(
+      final long eventKey,
+      final Intent eventState,
+      final UnpackedObject eventValue,
+      final ValueType valueType,
+      final long requestId,
+      final int requestStreamId);
+
+  /**
+   * Appends a task to be executed after a successful commit ProcessingResultBuilder (replacement
+   * for side effects)
+   */
+  ProcessingResultBuilder appendPostCommitTask(Runnable r);
+
   /**
-   * Resets the processing result build to its initial states (removes all follow up records,
-   * responses and commit tasks
+   * Resets the processing result build to its initial states (removes all follow up records, the
+   * response and post commit tasks)
    */
-  void reset();
+  ProcessingResultBuilder reset();
```

Comment:
This is probably not so important, but it made me wonder so I wrote it down.

❓ Can you clarify what this method does and returns:
- it only returns a new, empty ProcessingResultBuilder without touching the existing ProcessingResultBuilder, so the caller can still access the original data in the original builder object,
- it resets the existing ProcessingResultBuilder but returns the cleared out data, so the cleared out data can be inspected by the caller,
- it resets the existing ProcessingResultBuilder and returns this empty ProcessingResultBuilder, just for chained calls.

Reply:
The last option.